Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm

```diff
@@ -3704,6 +3704,11 @@ void effectiveAppearanceDidChange(Inspector::ExtensionAppearance appearance) ove
     });
 }
 
+bool WebExtensionContext::hasDynamicOrSessionDeclarativeNetRequestRules()
+{
+    return !m_sessionRulesIDs.isEmpty() || !m_dynamicRulesIDs.isEmpty();
```

Comment:
This should call `extension().hasContentModificationRules() || …` too count the static rules.

Reply:
i think the intention here is that this method is for dynamic and sessions rules only

Reply:
discussed on slack and decided it was best to have a full proof method that considers the dynamic, session, and enabled static rules